### PR TITLE
[HDInsight DataPlane]-Add descriptions for sparkjob related definitions to make it clear

### PR DESCRIPTION
**Pull request by sunsw1994**
1. rename spark job sdk's test method 
2. add descriptions for sparkjob related definitions to make it clear
3. add JobState.cs to enum states of jobs

### CODE REVIEW

**Comment by sunsw1994** [comment]
hi @AlexGhiondea, could please help us to review this PR. Thanks.

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
-## 3.0.0-preview.1 (Unreleased)
-
+## 3.0.0-preview.2 (Unreleased)
```
Comment (aim-for-better) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/CHANGELOG.md:1]: Add `Release History` above this line.  Please refer other data plane sdk about the CHANGELOG.md

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
...
+- Support using Apache Spark REST API to submit remote jobs to an HDInsight Spark cluster.
+## 3.0.0-preview.1
```
Comment (aim-for-better) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/CHANGELOG.md:3]: Add a blank line here.

File: sdk/hdinsight/Microsoft.Azure.HDInsight.Job/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
-## 3.0.0-preview.1 (Unreleased)
-
+## 3.0.0-preview.2 (Unreleased)
+- Support using Apache Spark REST API to submit remote jobs to an HDInsight Spark cluster.
```
Comment (aim-for-better) [inline sdk/hdinsight/Microsoft.Azure.HDInsight.Job/CHANGELOG.md:2]: >Support using Apache Spark REST API to submit remote jobs to an HDInsight Spark cluster. [](start = 2, length = 88)

I think that this description is more suitable for version preview.1.
I suggest to modify this description based what you have changed.

**Review by aim-for-better — APPROVED** [review]
Some comments left.

**Comment by sunsw1994** [comment]
> Some comments left.

Update a new commit.

**Comment by sunsw1994** [comment]
hi @AlexGhiondea, could please help us to review and merge this PR. Thanks.

**Review by aim-for-better — APPROVED** [review]
LGTM

**Review by tg-msft — APPROVED** [review]
The changes look good

**Pull request merged**